### PR TITLE
don't blow up when json object has no recipes

### PR DESCRIPTION
**Pull request by kisoku**
fixes loading brand new node objects from json

### CODE REVIEW

**Comment by coderanger** [comment]
We should add a regression test for this.

File: lib/chef/node.rb [hunk 1]
```diff
@@ -546,7 +546,7 @@ def self.json_create(o)
       if o.has_key?("run_list")
         node.run_list.reset!(o["run_list"])
       else
-        o["recipes"].each { |r| node.recipes << r }
+        o["recipes"].each { |r| node.recipes << r } if o.has_key?("recipes")
```
Comment (thommay) [inline lib/chef/node.rb]: seems like the logic would be better as part of the parent `if` statement:
```ruby
elsif o.has_key?("recipes")
...
```

**Comment by kisoku** [comment]
@thommay diff updated as per your comments

**Comment by kisoku** [comment]
@coderanger I can work on some tests, but unsure as to how to handle the json fixtures

**Comment by lamont-granquist** [comment]
:+1:

**Comment by smurawski** [comment]
:+1:

**Comment by thommay** [comment]
I'm :+1:, think we should just go ahead and merge

**Pull request merged**